### ISSUE TITLE
Many-to-One unique relationship

### ISSUE THREAD

**Issue by Floriferous** [Herteby/denormalize#17]
I'm trying to get a cache working where the relationship is "many-to-one unique", but I think this package doesn't handle it. Here's an example:

I have a collection of Authors, and a collection of Comments. The authors store an array of `commentId`s, and I want each comment to denormalize the author, without it being an array.

```js
Authors // { commentIds: ['commentId1', 'commentId2'] }

// What I want
Comments: // { authorCache: { _id: 'authorId', name: 'John' } }

// What I get
Comments: // { authorCache: [{ _id: 'authorId', name: 'John' }] }
```

Usually many-to-one means there can be many authors linking to the same comment, but that's not the case here, so the cache is not correct.

**Comment by doronrk** [comment]
Yeah I agree that it would be nice to be able to express "unique" here. I'm running into this as well using grapher with denormalize.

**Comment by doronrk** [comment]
@Herteby is this something you would accept a PR for?

**Comment by Herteby** [comment]
Not 100% if @-mentioning people in other threads means they get notified, so posting a link here too:
https://github.com/Herteby/denormalize/issues/21
